corehq/apps/users/management/commands/clone_web_users.py

```diff
@@ -0,0 +1,215 @@
+import csv
+import logging
+
+from django.contrib.auth.models import User
+from django.core.management.base import BaseCommand
+from django.template.loader import render_to_string
+from django.utils.translation import ugettext_lazy as _
+
+from toggle.shortcuts import set_toggle
+
+import settings
+from corehq.apps.export.dbaccessors import _get_export_instance
+from corehq.apps.export.models import ExportInstance
+from corehq.apps.hqwebapp.tasks import send_html_email_async
+from corehq.apps.saved_reports.models import ReportConfig, ReportNotification
+from corehq.apps.users.models import CouchUser, DomainMembership, WebUser
+from corehq.const import USER_CHANGE_VIA_CLONE
+from corehq.toggles import toggles_enabled_for_user
+
+logger = logging.getLogger(__name__)
+
+OLD_USERNAME = 'old_username'
+NEW_USERNAME = 'new_username'
+
+
+class Command(BaseCommand):
+    help = """
+    Creates new web users with the same data as existing web users
+    - Input is a CSV file with old_username and new_username columns
+    - The new user is created, data is copied/transferred over, and the old user is deactivated which will log
+    users out on their next request
+    - Sends an email to the new user's preferred email and old user's preferred email informing them of the change
+    If a user wants to reactivate their old account, they can request this via support who has the ability to make
+    this change in hq admin  under User Administration -> Lookup user by email -> Disable/Enable User Account
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument('file', help='')
+        parser.add_argument('--verbose', action="store_true")
+
+    def handle(self, file, **options):
+        logger.setLevel(logging.INFO if options["verbose"] else logging.WARNING)
+        for old_username, new_username in iterate_usernames_to_update(file):
+            logger.info(f'Cloning old user {old_username} to new user {new_username}')
+            try:
+                old_user, new_user = clone_user(old_username, new_username)
+            except CouchUser.Inconsistent as e:
+                logger.error(f'{e}\nSkipping the pair ({old_username}, {new_username})')
+                continue
+
+            deactivate_django_user(old_user.get_django_user())
+            send_deprecation_email(old_user, new_user)
+
+
+def iterate_usernames_to_update(file):
+    with open(file, newline='') as csvfile:
```

Comment:
nit: (I must have missed this on my first pass, but not a blocker). I don't think `csv` readers care what flavor of universal newlines the file is opened in, so the `newline=''` can be omitted. Setting the newline mode can lead to confusion since it appears like a non-standard CSV format might be expected, which I don't think is the case here.